erdpy/contracts.py

```diff
@@ -177,16 +181,33 @@ def _interpret_return_data(self, data: str) -> Any:
 
 
 def _prepare_argument(argument: Any):
-    as_string = str(argument).upper()
-
-    if as_string.startswith(HEX_PREFIX):
-        return _prepare_hexadecimal(as_string)
-
-    return _prepare_decimal(as_string)
+    as_str = str(argument)
+    as_hex = _to_hex(as_str)
+    return as_hex
+
+
+def _to_hex(arg: str):
+    if arg.startswith(HEX_PREFIX):
+        return _prepare_hexadecimal(arg)
+
+    if arg.isnumeric():
+        return _prepare_decimal(arg)
+    elif arg.startswith(ERD_BECH32_PREFIX):
```

Comment:
This will require `address:...` in the future, as well.

Reply:
`address:` in the mandos format currently generates dummy addresses from simple strings. I would rather not mix them.
I think for now this PR is fine as it is.

Reply:
cool, feel free to merge then :)

Reply:
@mateigraura, not all commits are signed. Do you think you can redo the PR using signed commits?

If not possible, let us know - and we will temporarily deactivate the "all signed commits" constraint.

Reply:
I'd rather not, would be best if this one gets merged so the discussion remains in the merged PRs history

Reply:
Next PRs should contain only signed commits. Otherwise we won't merge them as it is a security risk.